src/utils/intersection-observer-polyfill.js

```diff
@@ -14,6 +14,7 @@
  * limitations under the License.
  */
 
+import {MessageType} from '../3p-frame-messaging';
```

Comment:
Should we also rename this file/class to include "Legacy"?

Reply:
I wouldn't call it legacy because `IntersectionObserverHostApi` supports the `send-intersections` API for iframes. And we plan to continue the support there. 

This file also contains some utility functions to generate intersectionObserverEntry. I agree `intersection-observer-polyfill` is not not a good file name. 

What about I can switch the file name to `intersection-observer-entry-helper.js` and then move the `IntersectionObserverHostApi` class to `iframe-helper.js`?

Reply:
How about we just call it then actually `intersection-observer-3p-host` or such?

Reply:
I like it! Will rename